### ISSUE TITLE
Has anyone gotten this working on Microsoft Azure? If so what ports should we be listening on and what headers should we remove?

### ISSUE THREAD

**Issue by uhlenkott** [Rob--W/cors-anywhere#82]

**Comment by cadilhac** [comment]
Wondering also... Just installed it on Azure and I get a 500 error...

**Comment by cadilhac** [comment]
I got it finally working. Azure was selecting a very old node version...
However I still have a minor issue. This requests creates error 500 and no error appears in the logs:

`https://myserver.azurewebsites.net/http://www.remoteserver.com/query=john`

This one runs fine:

`https://myserver.azurewebsites.net/www.remoteserver.com/query=john`

**Comment by Rob--W** [comment]
@cadilhac [Your question on SO](https://stackoverflow.com/questions/34197132/azure-websites-throw-500-error-when-there-is-a-colon-in-the-url) has been answered, so I will add another note: you were using CORS Anywhere on a Windows server, where `:` is disallowed by default in a path name (this is probably a mitigation to avoid vulnerabilities surrounding alternate data streams, CVE-1999-0278). The recommended solution is to run CORS Anywhere on Linux (or if you want to stick to Windows+IIS, allow `:` in IIS, but after doing so, verify that you aren't introducing a security vulnerability by doing so).

Lastly, note that omitting "http://" is not equivalent to `http:`. If the scheme is not specified, CORS Anywhere will default to the scheme that was used for the original request, `https` in the case of `https://myserver.azurewebsites.net/www.remoteserver.com/query=john`

**Comment by cadilhac** [comment]
Thanks Rob.

And if my web site is http, but I use

`http://myserver.azurewebsites.net/www.remoteserver.com:443/query=john`

it will do the trick, right?

**Comment by Rob--W** [comment]
Yes. But there is a colon in your URL again.

**Comment by cadilhac** [comment]
Yes, but strangely, it works. Are you sure this is not ':/' that is forbidden?

**Comment by chrischip** [comment]
I built a docker container with nothing except cors-anywhere, using node 10.14.2 as base. I run that locally and it works as expected.

I then upload that docker container to a container registry, ask Azure to Webapp (linux based) to use that container. It deployed successfully and seems to be running. However, when it tried to serve any request, it always return 404. Looks like it is Azure outbound network issue.....Any suggestion on how to get this fixed? Thank you?

**Comment by chrischip** [comment]
Please look at the PR if you want to deploy this to Azure. This patch will make it work there

#146

**Comment by jamiemaguiredotnet** [comment]
Has anybody got this to work in Azure yet? I have the same issue.

**Comment by chrischip** [comment]
> Has anybody got this to work in Azure yet? I have the same issue.

my PR was rejected. However you are welcome to look at my change and fork your own branch...

https://github.com/Rob--W/cors-anywhere/pull/146/commits/049bb95ec578c3d80d8cc72ca4fbd556f74808ce
